outbound-dialer/src/test/java/org/motechproject/nms/outbounddialer/it/TargetFileServiceBundleIT.java

```diff
@@ -102,20 +111,33 @@ private void setupDatabase() {
 
 
     @Test
-    public void testTargetFileGeneration() {
-        SettingsFacade settingsFacade = settingsService.getSettingsFacade();
-
-        String oldNotificationUrl = settingsFacade.getProperty("outbound-dialer.target_file_notification_url");
-        settingsFacade.setProperty("outbound-dialer.target_file_notification_url", "http://xxx.yyy/zzz");
+    public void testTargetFileGeneration() throws NoSuchAlgorithmException, IOException {
         setupDatabase();
         TargetFileNotification tfn = targetFileService.generateTargetFile();
-        settingsFacade.setProperty("outbound-dialer.target_file_notification_url", oldNotificationUrl);
         assertNotNull(tfn);
 
         // Should not pickup subscription22 because its status is COMPLETED
         assertEquals(5, (int) tfn.getRecordCount());
 
-        //todo: verify tfn data actually matches created file
+        //read the file to get checksum & record count
+        File homeDir = new File(System.getProperty("user.home"));
+        File targetDir = new File(homeDir,
+                settingsService.getSettingsFacade().getProperty("outbound-dialer.target_file_directory"));
+        File targetFile = new File(targetDir, tfn.getFileName());
+        MessageDigest md = MessageDigest.getInstance("MD5");
+        int recordCount = 0;
+        try (InputStream is = Files.newInputStream(targetFile.toPath());
+             BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
+            DigestInputStream dis = new DigestInputStream(is, md);
```

Comment:
Is this correct?  I see you instantiate dis but it doesn't look like it is used.